user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -272,13 +274,14 @@ def _verify_exec_cluster(self):
         self._handle_non_running_exec_cluster(msg)
 
     def launch(self):
-        self._init_tool()
-        self._connect_to_execution_cluster()
-        self._process_arguments()
-        self._execute()
-        self._collect_result()
-        self._archive_phase()
-        self._finalize()
+        with Loader(ToolLogging.is_debug_mode_enabled()):
+            self._init_tool()
+            self._connect_to_execution_cluster()
+            self._process_arguments()
+            self._execute()
+            self._collect_result()
+            self._archive_phase()
+            self._finalize()
```

Comment:
Filed a new issue #559 to track this as it increases the scope.

Reply:
Note that the `_finalize()` phase disabled the epilogue to prevent mixing between the stdout and the logging

```
    @phase_banner('Generating Report Summary',
                  enable_epilogue=False)
```